test/unit/test_illumina.py

```diff
@@ -150,6 +150,29 @@ def test_tarball_fail_missing_data(self):
             self.assertRaises(Exception, idir.get_RunInfo())
 
 
+class TestDifficultSampleNames(TestCaseWithTmp):
+    
+    def test_paired_1(self):
+        inDir = util.file.get_test_input_path(self)
+        outBam = util.file.mkstempfname('.bam')
+        outHeader = util.file.mkstempfname('.txt')
+        sampleSheet = os.path.join(inDir, 'SampleSheet.csv')
+        runInfo = os.path.join(inDir, 'RunInfo.xml')
+        fastq = (os.path.join(inDir, 'mebv-0-1_S5_L001_R1_001.fastq.gz'),
+                 os.path.join(inDir, 'mebv-0-1_S5_L001_R2_001.fastq.gz'))
+        illumina.miseq_fastq_to_bam(outBam, sampleSheet, fastq[0], inFastq2=fastq[1], runInfo=runInfo)
+        rgs = list(tools.samtools.SamtoolsTool().getReadGroups(outBam).values())
+        self.assertEqual(len(rgs), 1)
+        rgs = rgs[0]
+        self.assertEqual(rgs.get('ID'), 'AEF96')
+        self.assertEqual(rgs.get('PL'), 'illumina')
+        self.assertEqual(rgs.get('PU'), 'AEF96.1.CGTACTAG-CTAAGCCT')
+        # we care more about Picard not failing, but if we care about how filenames are processed:
+        #self.assertEqual(rgs.get('LB'), 'difficult-value+for_-sample_name0.1')
+        #self.assertEqual(rgs.get('SM'), 'difficult-value+for_-sample_name0.1')
```

Comment:
Any reason not to include these tests? At a minimum, it'd be nice to test that the fields exist and are non-empty.